module/solutions/helper.js

```diff
@@ -142,4 +1012,2537 @@ module.exports = class SolutionsHelper {
+          //   checkForTargetedSolution.result.surveyId = surveySubmissionData[0]
+          //     .surveyId
+          //     ? surveySubmissionData[0].surveyId
+          //     : "";
+          //   checkForTargetedSolution.result.submissionStatus =
+          //     surveySubmissionData[0].status;
+          // } else if (!isSolutionActive) {
+          //   throw new Error(CONSTANTS.apiResponses.LINK_IS_EXPIRED);
+          // }
+        } else if (solutionData.type === CONSTANTS.common.IMPROVEMENT_PROJECT) {
+          // Targeted solution
+          if (
+            checkForTargetedSolution.result.isATargetedSolution &&
+            createProject
+          ) {
+            //targeted user with project creation
+
+            let projectDetailFromLink =
+              await projectQueries.projectDocument(
+                {"solutionId" : solutionId}
+              );
+
+            if (!projectDetailFromLink || !projectDetailFromLink.data) {
+              return resolve(projectDetailFromLink);
+            }
+            if (projectDetailFromLink.data.length < 1 && !isSolutionActive) {
+              throw new Error(CONSTANTS.apiResponses.LINK_IS_EXPIRED);
+            }
+
+            checkForTargetedSolution.result["projectId"] = projectDetailFromLink
+              .data._id
+              ? projectDetailFromLink.data._id
+              : "";
+          } else if (
+            checkForTargetedSolution.result.isATargetedSolution &&
+            !createProject
+          ) {
+            //targeted user with no project creation
+            let findQuery = {
+              userId: userId,
+              projectTemplateId: solutionData.projectTemplateId,
+              referenceFrom: {
+                $ne: CONSTANTS.common.LINK,
+              },
+              isDeleted: false,
+            };
+
+            let checkTargetedProjectExist =
+              await projectQueries.projectDocument(
+                userToken,
+                findQuery,
+                ["_id"]
+              );
+
+            if (
+              checkTargetedProjectExist.success &&
+              checkTargetedProjectExist.data &&
+              checkTargetedProjectExist.data.length > 0 &&
+              checkTargetedProjectExist.data[0]._id != ""
+            ) {
+              checkForTargetedSolution.result["projectId"] =
+                checkTargetedProjectExist.data[0]._id;
+            } else if (!isSolutionActive) {
+              throw new Error(CONSTANTS.apiResponses.LINK_IS_EXPIRED);
+            }
+          } else {
+            if (!isSolutionActive) {
+              throw new Error(CONSTANTS.apiResponses.LINK_IS_EXPIRED);
+            }
+
+            // check if private-Project already exists
+            let checkIfUserProjectExistsQuery = {
+              createdBy: userId,
+              referenceFrom: CONSTANTS.common.LINK,
+              link: link,
+            };
+            let checkForProjectExist =
```

Comment:
@Prajwal17Tunerlabs you dont need to pass the token and you dont need to check the success also